akka-remote/src/main/scala/akka/remote/UniqueAddress.scala

```diff
@@ -0,0 +1,17 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote
+
+import akka.actor.Address
+
+@SerialVersionUID(1L)
+final case class UniqueAddress(address: Address, uid: Int) extends Ordered[UniqueAddress] {
+  override def hashCode = uid
+
+  def compare(that: UniqueAddress): Int = {
```

Comment:
long term we can replace `akka.cluster.UniqueAddress` with this one, and that's why I included the ordering